src/factories/TransferProxyFactory.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.21;
+
+import {SafeTransferLib} from "src/libraries/SafeTransferLib.sol";
+import {IPoolManager} from "src/interfaces/IPoolManager.sol";
+import {ITransferProxy, ITransferProxyFactory} from "src/interfaces/factories/ITransferProxy.sol";
+
+contract TransferProxy is ITransferProxy {
+    IPoolManager public immutable poolManager;
+    bytes32 public immutable destination;
+    address public immutable recoverer;
+
+    constructor(address poolManager_, bytes32 destination_, address recoverer_) {
+        poolManager = IPoolManager(poolManager_);
+        destination = destination_;
+        recoverer = recoverer_;
+    }
+
+    /// @inheritdoc ITransferProxy
+    function transfer(address asset, uint128 amount) external {
```

Comment:
What do you think about adding a `destination` argument here as well, and revert if it's not equal the set destination on the contract as a sanity check. Otherwise I dont think many people will actually verify the destination before transferring.

Reply:
I don't think that's worth the gas cost, should just be abstracted away in the UI